server/sonar-server/src/main/java/org/sonar/server/computation/ws/CeProjectWsAction.java

```diff
@@ -0,0 +1,93 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.ws;
+
+import java.util.List;
+import org.apache.ibatis.session.RowBounds;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.web.UserRole;
+import org.sonar.core.util.Uuids;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.ce.CeActivityDto;
+import org.sonar.db.ce.CeActivityQuery;
+import org.sonar.db.ce.CeQueueDto;
+import org.sonar.server.user.UserSession;
+import org.sonar.server.ws.WsUtils;
+import org.sonarqube.ws.WsCe;
+
+/**
+ * GET api/ce/project
+ * <p>Get the pending and last executed tasks of a given project</p>
+ */
+public class CeProjectWsAction implements CeWsAction {
+
+  public static final String PARAM_COMPONENT_ID = "componentId";
+
+  private final UserSession userSession;
+  private final DbClient dbClient;
+  private final CeWsTaskFormatter formatter;
+
+  public CeProjectWsAction(UserSession userSession, DbClient dbClient, CeWsTaskFormatter formatter) {
+    this.userSession = userSession;
+    this.dbClient = dbClient;
+    this.formatter = formatter;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller.createAction("project")
+      .setDescription("Get the pending and last executed tasks of a given project")
+      .setInternal(true)
+      .setResponseExample(getClass().getResource("CeProjectWsAction/example.json"))
+      .setHandler(this);
+
+    action.createParam(PARAM_COMPONENT_ID)
+      .setRequired(true)
+      .setExampleValue(Uuids.UUID_EXAMPLE_01);
+  }
+
+  @Override
+  public void handle(Request wsRequest, Response wsResponse) throws Exception {
+    String componentUuid = wsRequest.mandatoryParam(PARAM_COMPONENT_ID);
+    userSession.checkProjectUuidPermission(UserRole.USER, componentUuid);
+
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      List<CeQueueDto> queueDtos = dbClient.ceQueueDao().selectByComponentUuid(dbSession, componentUuid);
+      CeActivityQuery activityQuery = new CeActivityQuery()
+        .setComponentUuid(componentUuid)
+        .setOnlyCurrents(true);
+      List<CeActivityDto> activityDtos = dbClient.ceActivityDao().selectByQuery(dbSession, activityQuery, new RowBounds(0, 1));
+
+      WsCe.ProjectResponse.Builder wsResponseBuilder = WsCe.ProjectResponse.newBuilder();
+      wsResponseBuilder.addAllQueue(formatter.formatQueue(dbSession, queueDtos));
+      if (activityDtos.size() == 1) {
```

Comment:
shouldn't we already manage when there'll be several analysis at the same time ?
or we should:
- throw an exception
- clean the db to have only one current activity
- another solution I haven't thought of